Rhodanobacter LGT COG Abundance Plot

Analysis of data sets to identify overabundant COG's in individual Rhodanobacter genomes and in the Rhodanobacter genus pangenome.

Raw Data Files
	COG Abundance Files from IMG/er for Rhodanobacter and All Bacteria (as of Sept 5, 2014) using the 
		Compare Genomes->Abundance Profiles->Overview (All) with Matrix/Gene Count/Include All Rows options
		to ensure that every abundance file is the same length (ie includes all of the same COGs).
	
		Rhodanobacter_abund.txt - (16 species)
			Rhodanobacter only
		abund1.txt - (320 species)
			Acidobacteria
			Acetothermia
			Aerophobetes
			Aminicenantes
			Aquificae
			Armatimonadetes
			Atribacteria
			Actinobacteria (partial)
				Acidimicrobiales
				Bifidobacterales
				Coriobacteriales
				Nitriliruptorales
				Rubrobacterales
				Solirubrobacterales
				Unclassified
		abund2.txt - (878 species)
			Actinobacteria (partial)
				Actinomyceteales (partial)
					Acidothermaceae
					Actinomycetaceae
					Actinopolysporaceae
					Actinospicaceae
					Beutenbergiaceae
					Bagoriellaceae
					Brevibacteriaceae
					Catenulisporaceae
					Cellulomonadaceae
					Corynebacteriaceae
					Cryptosporangiaceae
					Dermabacteraceae
					Dermcoccaceae
					Dermatophilaceae
					Dietziaceae
					Frankiaceae
					Geodermatophilaceae
					Glycomycetaceae
					Gordoniaceae
					Intrasporangiaceae
					Jiangellaceae
					Jonesiaceae
					Kineosporiaceae
					Microbacteraceae
					Microbacteriaceae
					Micrococcaceae
					Micromonosporaceae
					Nakamurellaceae
					Nocardiaceae
					Nocardiopsaceae
					Promicromonosporaceae
					Propionibacteriaceae
					Pseudonocardiaceae
					Ruaniaceae
					Sanguibacteraceae
					Segniliparaceae
					Sporichthyaceae
					Streptosporangiaceae
					Thermomonosporaceae
					Tsukamurellaceae
					Unclassified
		abund3.txt - (990 species)
			Actinobacteria (partial)
				Actinomyceteales (partial)
					Mycobacteriaceae
					Streptomycetaceae
		abund4.txt - (941 species)
			BRC1
			Bacteroidetes
			Caldiserica
			Caldithrixae
			Calescamentes
			Candidatus Saccharibacteria
			Chlamydiae
		abund5.txt - (903 species)
			Chlorobi
			Chloroflexi
			Chrysiogenetes
			Cloacimonetes
			Cyanobacteria
			Deferribacteres
			Deinococcus-Thermus
			Dictyoglomi
			EM3
			Elusimicrobia
			Fervidibacteria
			Fibrobacteres
			Fusobacteria
			Gemmatimonadetes
			Gracilibacteria
			Hydrogenedentes
			Ignavibacteria
			Ignavibacteriae
			Latescibacteria
			Lentisphaera
			Marinimicrobia
			Microgenomates
			Nitrospirae
			Omnitrophica
			PER
			Parcubacteria
			Planctomycetes
			Poribacteria
			Synergisetes
			Tectomicrobia
			Tenericutes
			Thermodesulfobacteria
			Thermotogae
			Verrucomicrobia
			WS1
			candidate division JS1
			candidate division NC10
			candidate division SR1
			candidate division TM6
			unclassified
		abund6.txt - (944 species)
			Firmicutes (partial)
				Clostridia
				Erysipelotrichi
				Erysipelotrichia
				Unclassified
		abund7.txt - (498 species)
			Firmicutes (partial)
				Negativicutes
				Bacilli (partial)
					Lactobacillus (partial)
						Aerococcaceae
						Carnobacteriaceae
						Lactobacilliaceae
						Leuconostocaceae
		abund8.txt - (674 species)
			Firmicutes (partial)
				Bacilli (partial)
					Lactobacillus (partial)
						Enterococcaceae
		abund9.txt - (890 species)
			Firmicutes (partial)
				Bacilli (partial)
					Lactobacillus (partial)
						Lactococcus (partial)
							Streptococcus (partial)
								Through Pseodoporinus
		abund10.txt - (471 species)
			Firmicutes (partial)
				Bacilli (partial)
					Lactobacillus (partial)
						Lactococcus (partial)
							Streptococcus (partial)
								Remaining
		abund11.txt - (820 species)
			Firmicutes (partial)
				Bacillu (partial)
					Bacillales (partial)
						Alicyclobacillaceae
						Bacilliaceae
						Bacillales Family X: Incertae Sedis
						Listeriaceae
						Pasteuriaceae
						Planococcaceae
						Sporolactobacillaceae
						Thermoactinomyceaceae
						Unclassified
						Staphylococcaceae (partial)
							All but Staphylococcus
		abund12.txt - (437 species)
			Spirochaetes
		abund13.txt - (910 species)
			Firmicutes (partial)
				Bacilli (partial)
					Bacillales (partial)
						Staphylococcaceae (partial)
							Staphylococcus (partial)
								Through Staphylococcus aureus M0832
		abund14.txt - (727 species)
			Firmicutes (partial)
				Bacilli (partial)
					Bacillales (partial)
						Staphylococcaceae (partial)
							Staphylococcus (partial)
								From Staphylococcus aureus M0834 to T89467
		abund15.txt - (592 species)
			Firmicutes (partial)
				Bacilli (partial)
					Bacillales (partial)
						Staphylococcaceae (partial)
							Staphylococcus (partial)
								Remaining
		abund16.txt - (774 species)
			Proteobacteria (partial)
				Deltaproteobacteria
				Epsilonproteobacteria
				Zetaproteobacteria
				Unclassified
		abund17.txt - (993 species)
			Proteobacteria (partial)
				Alphaproteobacteria (partial)
					Caulobacterales
					Kiloniellales
					Kordiimonadales
					Magnetococcales
					Parvularculales
					Rhizobiales
					Sneathiellales
					Sphingomonales
					Unclassified
		abund18.txt - (403 species)
			Proteobacteria (partial)
				Alphaproteobacteria (partial)
					Rhodobacterales
					Rhodospirillales
					Rickettsiales
		abund19.txt - (859 species)
			Proteobacteria (partial)
				Betaproteobacteria
		abund20.txt - (887 species)
			Proteobacteria (partial)
				Gammaproteobacteria (partial)
					Acidithiobacillales
					Aeromonadales
					Alteromonadales
					Cardiobacteriales
					Legionellales
					Methylococcales
					Oceanospirillales
					Orbales
					Pasteurallales
					Salinisphaerales
					Thiotrichales
					Unclassified
		abund21.txt - (667 species)
			Proteobacteria (partial)
				Gammaproteobacteria (partial)
					Vibrionales
					Xanthomonadales
		abund22.txt - (965 species)
			Proteobacteria (partial)
				Gammaproteobacteria (partial)
					Moraxellaceae
					Unclassified
		abund23.txt - (673 species)
			Proteobacteria (partial)
				Gammaproteobacteria (partial)
					Pseudomonadales (partial)
						Pseudomonadaceae
		abund24.txt - (800 species)
			Proteobacteria (partial)
				Gammaproteobacteria (partial)
					Enterobacteriales (partial)
						All genera but Eschericia and Salmonella
		abund25.txt - (565 species)
			Proteobacteria (partial)
				Gammaproteobacteria (partial)
					Enterobacteriales (partial)
						Salmonella
		abund26.txt - (923 species)
			Proteobacteria (partial)
				Gammaproteobacteria (partial)
					Enterobacteriales (partial)
						Escherichia (partial)
							Through E. coli KOEGE77(2021)(B)
		abund27.txt - (867 species)
			Proteobacteria (partial)
				Gammaproteobacteria (partial)
					Enterobacteriales (partial)
						Escherichia (partial)
							Remaining Escherichia
							
Scripts

	CalcOddsRatios.R (R)
		Inputs
			files - vector of raw abundance files (IMG format) to be concatenated together
			sample - raw sample abundance file (IMG format)
			cogs - vector of COG ids (first column in files/sample)
			group - option to collapse all sample data to a single sample and add it to the end of the sample data
				(e.g. collapse species to single sample and analyze as genus) (default = "yes")
			log - return output as ln(odds ratio) instead of odds ratio (default = "yes")
		Dependencies
			bstats (for odds ratio calculation)
			data.table (for data reading)

	calc_odds.pl (Perl)
		Inputs
			infile - sample COG abundance input file (IMG format)
			outfile - output file for odds ratio results
			sample_included - "yes" or "no" depending on if sample data is included in the input abundance data
		Dependencies
			Statistics::ChisqIndep
			Text::NSP::Measures::2D::Fisher::right
			Getopt::Long

Processed Data Files

Experimental Design

	1)	Load COG abundance files into R as data.frames using data.table
	2)	Concatenate abundance objects into a single data.frame containing all of the abundance information
	3)	Add abundance data for Rhodanobacter genera and FRC metagenomes
	4)	Convert abundance data to numerical matrix
	5)	Sum rows and columns to get abundances for each COG in all bacteria and all COGs in each bacteria
	6)	Sum to get all COGs in all bacteria
	7)	Calculate ln(odds ratios) for all COGs in each bacteria
	8)	chi square test to identify significantly overabundant COGs
	9)	Select desired COGs
	10)	Create heat map using vegan heatmap2
